rust/kcl-lib/src/execution/artifact.rs

```diff
@@ -215,6 +215,19 @@ pub struct StartSketchOnPlane {
     pub code_ref: CodeRef,
 }
 
+#[derive(Debug, Clone, Serialize, PartialEq, ts_rs::TS)]
+#[ts(export_to = "Artifact.ts")]
+#[serde(rename_all = "camelCase")]
+pub struct SketchBlock {
+    pub id: ArtifactId,
+    /// The plane ID if the sketch block is on a specific plane, None if it's on a default plane.
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub plane_id: Option<ArtifactId>,
+    pub code_ref: CodeRef,
+    /// The sketch ID (ObjectId) for the sketch scene object.
+    pub sketch_id: usize,
+}
```

Comment:
I assume we want a new artifact for the sketch block, How's this for a first pass. I'm guessing we're going to want to add segment artifacts too and link them from here.

so far the plane_id isn't doing what I want it to, I'm a bit too tired to write out why, but I think it should be in teh struct either way.